src/components/date-picker/date-picker.tsx

```diff
@@ -462,9 +504,10 @@ export class DatePicker implements LocalizedComponent {
         <calcite-date-picker-month-header
           activeDate={activeDate}
           headingLevel={this.headingLevel || HEADING_LEVEL}
-          intlNextMonth={this.intlNextMonth}
-          intlPrevMonth={this.intlPrevMonth}
-          intlYear={this.intlYear}
+          intlNextMonth={this.messages.nextMonth}
```

Comment:
I think this is fine to keep (month-header not being a localized/t9n component) since it's a supporting component that won't be used outside of this context. Maybe we can add a note here?

Reply:
I think that's given considering it is not documented as a public component.

Reply:
We could make it clearer by adding a note under conventions/t9n module about internal components only needing to provide a prop to forward messages (`messages`?).